src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,200 @@
+package com.conveyal.datatools.manager.jobs;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.auth.Auth0UserProfile;
+import com.conveyal.datatools.manager.controllers.api.DeploymentController;
+import com.conveyal.datatools.manager.models.Deployment;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.models.OtpServer;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Auto deploy new feed version to OTP server if {@link Project#autoDeploy} is enabled and other conditions are met
+ * (e.g., feed version has no critical errors, active deployment is not in progress, there are no other feed fetches in
+ * progress, etc.). This job must run after {@link ValidateFeedJob} as it has a dependency on the outcome of
+ * {@link FeedVersion#hasCriticalErrors}.
+ */
+public class AutoDeployJob extends MonitorableJob {
+    public static final Logger LOG = LoggerFactory.getLogger(AutoDeployJob.class);
+
+    /**
+     * Projects to be considered for auto deployment.
+     */
+    private final List<Project> projects;
```

Comment:
I think this can just be a single project, since there is no use of the constructor the calls this on all of the projects. That would make the jobLogic a little nicer/less nested as well, and change the `continue` statements to `return` which I think simplifies/clarifies this class a bit.

Reply:
Updated.